zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/BufferStats.java

```diff
@@ -40,50 +41,50 @@
     private int maxBufferSize = INIT_VALUE;
 
     /**
-     * Size of the last buffer usage.
+     * Updates statistics by setting the last buffer usage size.
+     *
+     * @param value The last buffer size; must be equal to or greater than 0
      */
-    public synchronized int getLastBufferSize() {
-        return lastBufferSize;
+    public synchronized void setLastBufferSize(final int value) {
+        this.lastBufferSize = value;
+        this.minBufferSize = minBufferSize < 0 ? value : Math.min(minBufferSize, value);
```

Comment:
This logic is based on the assumption that `INIT_VALUE` is negative. A bit fragile.